qiita_db/test/test_software.py

```diff
@@ -0,0 +1,170 @@
+               "analysis from raw DNA sequencing data")
+        self.assertEqual(Software(1).description, exp)
+
+    def test_commands(self):
+        self.assertEqual(Software(1).commands, [1, 2, 3])
+
+    def test_publications(self):
+        self.assertEqual(Software(1).publications,
+                         [['10.1038/nmeth.f.303', '20383131']])
+
+    def test_add_publications(self):
+        tester = Software(1)
+        self.assertEqual(tester.publications,
+                         [['10.1038/nmeth.f.303', '20383131']])
+        tester.add_publications([['10.1000/nmeth.f.101', '12345678']])
+        exp = [['10.1038/nmeth.f.303', '20383131'],
+               ['10.1000/nmeth.f.101', '12345678']]
+        self.assertItemsEqual(tester.publications, exp)
+
+
+@qiita_test_checker()
+class ParametersTests(TestCase):
+    def test_init(self):
+        obs = Parameters(1, Command(1))
+        self.assertEqual(obs.id, 1)
+        self.assertEqual(obs._table, "preprocessed_sequence_illumina_params")
+
+    def test_exists(self):
+        cmd = Command(1)
+        obs = Parameters.exists(
+            cmd, max_bad_run_length=3, min_per_read_length_fraction=0.75,
+            sequence_max_n=0, rev_comp_barcode=False,
+            rev_comp_mapping_barcodes=False, rev_comp=False,
+            phred_quality_threshold=3, barcode_type="golay_12",
+            max_barcode_errors=1.5)
+        self.assertTrue(obs)
+
+        obs = Parameters.exists(
+            cmd, max_bad_run_length=3, min_per_read_length_fraction=0.75,
+            sequence_max_n=0, rev_comp_barcode=False,
+            rev_comp_mapping_barcodes=False, rev_comp=False,
+            phred_quality_threshold=3, barcode_type="hamming_8",
+            max_barcode_errors=1.5)
+        self.assertFalse(obs)
+
+    def test_create(self):
+        cmd = Command(1)
+        obs = Parameters.create(
+            "test_create", cmd, max_bad_run_length=3,
+            min_per_read_length_fraction=0.75, sequence_max_n=0,
+            rev_comp_barcode=False, rev_comp_mapping_barcodes=False,
+            rev_comp=False, phred_quality_threshold=3,
+            barcode_type="hamming_8", max_barcode_errors=1.5)
+        self.assertEqual(obs.name, "test_create")
+
+        exp = {'max_bad_run_length': 3, 'min_per_read_length_fraction': 0.75,
+               'sequence_max_n': 0, 'rev_comp_barcode': False,
+               'rev_comp_mapping_barcodes': False, 'rev_comp': False,
+               'phred_quality_threshold': 3, 'barcode_type': "hamming_8",
+               'max_barcode_errors': 1.5}
+        self.assertEqual(obs.values, exp)
+
+    def test_iter(self):
+        cmd = Command(1)
+        obs = list(Parameters.iter(cmd))
+        exp = [Parameters(i, cmd) for i in range(1, 8)]
+        self.assertEqual(obs, exp)
+
+    def test_name(self):
+        self.assertEqual(Parameters(1, Command(1)).name, "Defaults")
+
+    def test_values(self):
+        exp = {'min_per_read_length_fraction': 0.75,
```

Comment:
Ok, can you add a comment to #1523?
Those tables already exist in the DB and the changes may be bigger that just changing those types here. Since this may impact qiita_ware and it will be revamped with the plugin system, I would suggest to add those changes later.

Thanks!